typescript/packages/plugins/velodrome-plugin/src/velodrome.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Tool } from "@goat-sdk/core";
+import type { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { parseUnits } from "viem";
+import { ROUTER_ABI } from "./abi/router";
+import { SwapExactTokensParams, AddLiquidityParams } from "./parameters";
+
+const ROUTER_ADDRESS = "0xa062aE8A9c5e11aaA026fc2670B0D65cCc8B2858";
+const POOL_FACTORY = "0xF1046053aa5682b4F9a81b5481394DA16BE5FF5a";
+
+export class VelodromeService {
+    @Tool({
+        name: "swap_exact_tokens",
+        description: "Swap an exact amount of tokens on Velodrome",
+    })
+    async swapExactTokens(walletClient: EVMWalletClient, parameters: SwapExactTokensParams) {
+        const deadline = Math.floor(Date.now() / 1000) + 60 * 20; // 20 minutes
+
+        const amountIn = parseUnits(parameters.amountIn, parameters.tokenInDecimals);
```

Comment:
Would not parse these here, let the LLM figure out when to use the convertToBaseUnits tool :)